planner/core/issuetest/planner_issue_test.go

```diff
@@ -118,3 +118,30 @@ func TestIssue42732(t *testing.T) {
 	tk.MustExec("INSERT INTO t2 VALUES (1, 1)")
 	tk.MustQuery("SELECT one.a, one.b as b2 FROM t1 one ORDER BY (SELECT two.b FROM t2 two WHERE two.a = one.b)").Check(testkit.Rows("1 1"))
 }
+
+// https://github.com/pingcap/tidb/issues/45036
+func TestIssue45036(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("CREATE TABLE ads_txn (\n" +
+		"  `cusno` varchar(10) NOT NULL,\n" +
+		"  `txn_dt` varchar(8) NOT NULL,\n" +
+		"  `unn_trno` decimal(22,0) NOT NULL,\n" +
+		"  `aml_cntpr_accno` varchar(64) DEFAULT NULL,\n" +
+		"  `acpayr_accno` varchar(35) DEFAULT NULL,\n" +
+		"  PRIMARY KEY (`cusno`,`txn_dt`,`unn_trno`) NONCLUSTERED\n" +
+		") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin\n" +
+		"PARTITION BY LIST COLUMNS(`txn_dt`)\n" +
+		"(PARTITION `p20000101` VALUES IN ('20000101'),\n" +
+		"PARTITION `p20220101` VALUES IN ('20220101'),\n" +
+		"PARTITION `p20230516` VALUES IN ('20230516'))")
+	tk.MustExec("analyze table ads_txn")
```

Comment:
Is this `analyze` unnecessary?

Reply:
Yes, to prepare global stats for dynamic prune.